utils/event.c

```diff
@@ -240,6 +240,21 @@ char * event_get_data_str(unsigned evt_id, void *data, bool verbose)
 	return str;
 }
 
+/**
+ * finish_events_file - cleanup memory for events in the given handle
+ * @handle: uftrace_data data structure
+ */
+void finish_events_file(struct uftrace_data *handle)
+{
+	struct uftrace_event *ev, *tmp;
+
+	list_for_each_entry_safe(ev, tmp, &handle->events, list) {
```

Comment:
Please call `list_del(&ev->list)` before releasing the entry.

Reply:
Thanks. It's updated again.

Reply:
Thanks, it'd be great if you can add a unit test to read a simple file and release the events.

Reply:
Thanks for waiting. I just added a unit test.